docs/subnets/customize-a-subnet.md

```diff
@@ -0,0 +1,708 @@
+
+```solidity
+// (c) 2022-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.8.0;
+
+interface NativeMinterInterface {
+    // Set [addr] to have the admin role over the minter list
+    function setAdmin(address addr) external;
+
+    // Set [addr] to be enabled on the minter list
+    function setEnabled(address addr) external;
+
+    // Set [addr] to have no role over the minter list
+    function setNone(address addr) external;
+
+    // Read the status of [addr]
+    function readAllowList(address addr) external view returns (uint256);
+
+    // Mint [amount] number of native coins and send to [addr]
+    function mintNativeCoin(address addr, uint256 amount) external;
+}
+```
+
+_Note: Both `ContractDeployerAllowList` and `ContractNativeMinter` can be used together._
+
+### Examples
+
+Subnet-EVM contains example contracts for precompiles under `/contract-examples`. It's a hardhat project with tests, tasks. For more information see [contract examples README](https://github.com/ava-labs/subnet-evm/tree/master/contract-examples#subnet-evm-contracts).
+
+
+
+## Chain Configs
+
+As described in [this doc](../nodes/maintain/chain-config-flags.md#subnet-chain-configs), each blockchain of subnets can have its own custom configuration. If a subnet's chain id is `2ebCneblahblahblah`, the config file for this chain is located at `{chain-config-dir}/2ebCneblahblahblah/config.json`.
+
+
+For blockchains created by or forked from Subnet-evm, most [C-Chain configs](../nodes/maintain/chain-config-flags.md#c-chain-configs) are applicable except [Avalanche Specific APIs](../nodes/maintain/chain-config-flags.md#enabling-avalanche-specific-apis).
+
+
+### Priority Regossip
+
+A transaction is "regossiped" when the node does not find the transaction in
+a block after `priority-regossip-frequency` (defaults to `1m`). By default, up to 16 transactions
+(max 1 per address) are regossiped to validators per minute.
+
+Operators can use "priority regossip" to more aggressively "regossip" transactions for a set of
+important addresses (like bridge relayers). To do so, you'll need to update your
+[chain config](../nodes/maintain/chain-config-flags.md#subnet-chain-configs) with the following:
+
+```json
+{
+  "priority-regossip-addresses": ["<YOUR 0x-ADDRESS>"]
+}
+```
+
+By default, up to 32 transactions from priority addresses (max 16 per address) are regossipped to validators per second.
+You can override these defaults with the following config:
+
+```json
+{
+  "priority-regossip-frequency": "1s",
+  "priority-regossip-max-txs": 32,
+  "priority-regossip-addresses": ["<YOUR 0x-ADDRESS>"],
+  "priority-regossip-txs-per-address": 16
+}
+```
+
+### `feeRecipient`
```

Comment:
Can we change this so that the header is just normal text instead of the highlighted version here?

Reply:
Done